topics/data-science/tutorials/snakemake/tutorial.md

```diff
@@ -0,0 +1,1714 @@
+---
+layout: tutorial_hands_on
+
+title: Make & Snakemake
+level: Intermediate
+zenodo_link: "https://zenodo.org/record/5562251"
+requirements:
+- type: "internal"
+  topic_name: data-science
+  tutorials:
+      - cli-basics
+      - cli-advanced
+follow_up_training: []
+
+questions:
+- What is Make & Snakemake
+- What is a Makefile/Snakefile
+- How do these improve pipelines over simple scripts?
+- Why is Snakemake better for scientific research and how can I use it
+- How do I use conda envs with Snakemake?
+objectives:
+- Write a snakefile that does a simple QC and Mapping workflow
+time_estimation: 3H
+key_points:
+- Make and Snakemake are ways to write pipelines in a declarative format
+- Instead of writing individual actions you wish to take, you describe how to produce each file you need, and the system executes those steps only if they're needed
+- If you're doing scientific research you should use Snakemake
+- Snakemake can significantly speed up your command line pipelines by providing automatic detection of inputs, and not re-creating them if they don't need
+- Snakemake also provides very easy access to paralellisation without complexity.
+- But it is more complex than using Galaxy
+subtopic: sciwms
+contributors:
+  - hexylena
+  - bazante1
+  - dirowa
+abbreviations:
+  SciWMS: Scientific Workflow Management System
+  DAG: Directed Acyclic Graph
+---
+
+Here you will learn to write both Make and Snakemake workflows. We teach two workflow engines because Snakemake uses a lot of the concepts of Make, and these concepts are somewhat complex and a very different way of thinking than you might be used to with workflow design.
+
+This tutorial is aimed at students and Galaxy community members who might want to convert Snakemake workflows into Galaxy workflows, but need to understand how Snakemake workflows work.
+
+> ### Agenda
```

Comment:
is this required, or need it more items?

Reply:
this is automatically filled for us by the framework